WooCommerce/Classes/ViewRelated/Coupons/Add and Edit Coupons/Coupon Text Validators/CouponAmountInputFormatter.swift

```diff
@@ -3,17 +3,25 @@ import Foundation
 /// `UnitInputFormatter` implementation for Coupon Amount input.
 ///
 struct CouponAmountInputFormatter: UnitInputFormatter {
+    let priceInputFormatter = PriceInputFormatter()
+
     func isValid(input: String) -> Bool {
         guard input.isNotEmpty else {
             return false
         }
-        return PriceInputFormatter().isValid(input: input)
+        return priceInputFormatter.isValid(input: input)
     }
 
     func format(input text: String?) -> String {
         guard text.isNilOrEmpty else {
-            return PriceInputFormatter().format(input: text)
+            return priceInputFormatter.format(input: text)
         }
         return "0"
     }
 }
+
+extension CouponAmountInputFormatter {
+    func value(from input: String) -> NSNumber? {
+        priceInputFormatter.value(from: input)
```

Comment:
Would it make sense to add unit tests for `CouponAmountInputFormatter` and `PriceInputFormatter`?

Reply:
Sure! Added tests in 135ca0af6a6eaf9875493fd07e3dd4b2f986ad74 and 84ae4565341c643e4df047a66728aca83d66d2b2.